2007-01-13  Jan Kratochvil  <jan.kratochvil@redhat.com>

	* lib/gdb.exp (get_compiler_info): Cope with nondefault output logs.


diff -u -rupN gdb-6.5-ppc/gdb/testsuite/lib/gdb.exp gdb-6.5/gdb/testsuite/lib/gdb.exp
--- gdb-6.5-ppc/gdb/testsuite/lib/gdb.exp	2007-01-12 14:33:26.000000000 -0500
+++ gdb-6.5/gdb/testsuite/lib/gdb.exp	2007-01-12 21:15:02.000000000 -0500
@@ -1351,9 +1351,10 @@ proc get_compiler_info {binfile args} {
 
     # Run $ifile through the right preprocessor.
     # Toggle gdb.log to keep the compiler output out of the log.
+    set log_old [log_file -info]
     log_file
     set cppout [ gdb_compile "${ifile}" "" preprocess [list "$args" quiet] ]
-    log_file -a "$outdir/$tool.log" 
+    eval "log_file $log_old"
 
     # Eval the output.
     set unknown 0


2007-01-14  Jan Kratochvil  <jan.kratochvil@redhat.com>

	* lib/gdb.exp: Abort on missing `site.exp'.


diff -u -rup gdb-6.5-orig/gdb/testsuite/lib/gdb.exp gdb-6.5/gdb/testsuite/lib/gdb.exp
--- gdb-6.5-orig/gdb/testsuite/lib/gdb.exp	2006-12-17 11:59:42.000000000 +0100
+++ gdb-6.5/gdb/testsuite/lib/gdb.exp	2007-01-14 16:27:03.000000000 +0100
@@ -22,6 +22,12 @@
 # need to be modified for any target, it can be done with a variable
 # or by passing arguments.
 
+if {$tool == ""} {
+    # Tests would fail, logs on get_compiler_info() would be missing.
+    send_error "`site.exp' not found, run `make site.exp'!\n"
+    exit 2
+}
+
 load_lib libgloss.exp
 
 global GDB
